src/App/Pages/Settings/LoginPasswordlessRequestsListViewModel.cs

```diff
@@ -0,0 +1,143 @@
+        }
+
+        public async Task RefreshAsync()
+        {
+            try
+            {
+                LoginRequestsList.ReplaceRange(await _authService.GetActivePasswordlessLoginRequestsAsync());
+
+                if (!LoginRequestsList.Any())
+                {
+                    Page.Navigation.PopModalAsync().FireAndForget();
+                }
+            }
+            catch (Exception ex)
+            {
+                HandleException(ex);
+            }
+        }
+
+        private async Task PasswordlessLoginAsync(PasswordlessLoginResponse request)
+        {
+            if (request.IsExpired)
+            {
+                await _platformUtilsService.ShowDialogAsync(AppResources.LoginRequestHasAlreadyExpired);
+                await Page.Navigation.PopModalAsync();
+                return;
+            }
+
+            var loginRequestData = await _authService.GetPasswordlessLoginRequestByIdAsync(request.Id);
+            if (loginRequestData.IsAnswered)
+            {
+                await _platformUtilsService.ShowDialogAsync(AppResources.ThisRequestIsNoLongerValid);
+                return;
+            }
+
+            var page = new LoginPasswordlessPage(new LoginPasswordlessDetails()
+            {
+                PubKey = loginRequestData.PublicKey,
+                Id = loginRequestData.Id,
+                IpAddress = loginRequestData.RequestIpAddress,
+                Email = await _stateService.GetEmailAsync(),
+                FingerprintPhrase = loginRequestData.RequestFingerprint,
+                RequestDate = loginRequestData.CreationDate,
+                DeviceType = loginRequestData.RequestDeviceType,
+                Origin = loginRequestData.Origin
+            });
+
+            await Device.InvokeOnMainThreadAsync(() => Application.Current.MainPage.Navigation.PushModalAsync(new NavigationPage(page)));
+        }
+
+        private async Task DeclineAllRequestsAsync()
+        {
+            try
+            {
+                if (!await _platformUtilsService.ShowDialogAsync(AppResources.AreYouSureYouWantToDeclineAllPendingLogInRequests, null, AppResources.Yes, AppResources.No))
+                {
+                    return;
+                }
+
+                await _deviceActionService.ShowLoadingAsync(AppResources.Loading);
+                var taskList = new List<Task>();
+                foreach (var request in LoginRequestsList)
+                {
+                    taskList.Add(_authService.PasswordlessLoginAsync(request.Id, request.PublicKey, false));
+                }
+                await Task.WhenAll(taskList);
+                await RefreshAsync();
+                await _deviceActionService.HideLoadingAsync();
+                _platformUtilsService.ShowToast("info", null, AppResources.RequestsDeclined);
+            }
+            catch (Exception ex)
+            {
+                HandleException(ex);
+            }
+        }
```

Comment:
i've been doing a couple of tests and if some of the tasks executing fails it will throw an error stating it. Since they all start at about the same time, and it is a very quick api call, it is very difficult to be in the middle of the execution. Nonetheless if that happens, maybe i the best course of action is to place a refresh call in the exception block?

Reply:
"Quick": it depends on the connection hehe. If the user has a really slow connection then you can definitely be in the middle of the execution.
Yes, I would refresh it on the exception block just in case